src/rdf4cpp/rdf/Literal.hpp

```diff
@@ -411,6 +411,105 @@ class Literal : public Node {
         return this->cast(IRI{T::datatype_id, node_storage}, node_storage);
     }
 
+    /**
+     * Tries to cast this literal to a literal of the given type and return the result without creating a Literal.
+     *
+     * @return conversion result
+     */
+    template<datatypes::LiteralDatatype T>
+    requires (!std::same_as<T, datatypes::xsd::String>)
+    std::optional<typename T::cpp_type> cast_to_value() const noexcept {
```

Comment:
Why does this exist?

Reply:
I think that would not work, because casting to a string is a edge case i purposefully avoid for cast_to_value.
(due to the borrowed/owned semantic and xsd::string::cpp_type not having an option for that)

Reply:
Forgot about that one, bummer